.gitignore

```diff
@@ -1 +1,2 @@
 bazel-*
+.DS_Store
```

Comment:
what is this?

Reply:
This is a macos-only file placed in every directory -- usually added when running `git add .` instead of something more fine-grained.

Reply:
correct - OSX is just adding these to every dir, so I exclude them by default to ensure they do not inadvertently make their way into the checked-in codebase.

Reply:
this is an OS specific artifact, and not one incurred by building the software, please remove it.

Reply:
it would be really nice to keep it since otherwise OSX users will see this and accidentally add the file repeatedly.

What are your guidelines for what should be allowed in .gitignore? This is a very common entry to add.

Reply:
FWIW this ignore isn't added to bazelbuild/bazel, bazelbuild/rules_kotlin, or bazelbuild/rules_go.